app/src/crypto.c

```diff
@@ -53,43 +53,36 @@ zxerr_t hash_sha224(uint8_t *input, uint16_t inputLen, uint8_t *output, uint16_t
     }
     cx_sha256_t ctx;
     cx_sha224_init(&ctx);
-    cx_hash(&ctx.header, CX_LAST, input, inputLen, output, 224);
+    cx_hash_no_throw(&ctx.header, CX_LAST, input, inputLen, output, 224);
     return zxerr_ok;
 }
 
 zxerr_t crypto_extractPublicKey(const uint32_t path[HDPATH_LEN_DEFAULT], uint8_t *pubKey, uint16_t pubKeyLen) {
     cx_ecfp_public_key_t cx_publicKey;
     cx_ecfp_private_key_t cx_privateKey;
-    uint8_t privateKeyData[32];
+    uint8_t privateKeyData[64] = {0};
```

Comment:
We also need to check if pubkey == NULL and hdpath is defined in this file. It's not necessary to have as param